### DISCUSSION TITLE
Mass of species

### DISCUSSION THREAD

**Discussion by DanRRRR** [epochpic/epoch#694] [Q&A]
Hello All, 
In the default case the  #DEFINES += $(D)PER_PARTICLE_CHARGE_MASS in Makefile is commented out and not used.

If it would be used then the mass of species can be found and written to output (say, by doing output in the file  diagnostics.f90/output_routines )  by defining the linked list and taking its element SpecieMass = current%mass.

But  how to find during output process the mass of species in diagnostics.f90/output_routines  in the default case ?

**Comment by Status-Mirror** [comment]
Hi @DanRRRR,

In a function like `diagnostics.f90/output_routines`, there are subroutines calls like `write_particle_variable`, which calls functions in `iterators.F90`. The `iterators.F90/it_output_real` function is responsible for cycling through the particles and writing specific variables to a file. Look at the iterators.F90 lines:

```
        part_m   = current_species%mass
```

and

```
        DO WHILE (ASSOCIATED(cur) .AND. (part_count < npoint_it))
          part_count = part_count + 1
#ifdef PER_PARTICLE_CHARGE_MASS
          array(part_count) = cur%mass
#else
          array(part_count) = part_m
#endif
          cur => cur%next
        END DO
```

These show you how to access the species mass when not running with `-DPER_PARTICLE_CHARGE_MASS`.

Hope this helps,
Stuart

**Reply by DanRRRR** [reply]
That's great, thanks, Stuart
